packages/framework/src/Model/Customer/User/CustomerUserFacade.php

```diff
@@ -297,7 +297,9 @@ protected function setEmail(string $email, CustomerUser $customerUser): void
 
         if (
             $customerUserByEmailAndDomain !== null
-            && $customerUser->getId() !== $customerUserByEmailAndDomain->getId()
+            && (
+                !$this->em->contains($customerUser) || $customerUser->getId() !== $customerUserByEmailAndDomain->getId()
```

Comment:
I could revert this code as we removed typehints from entities, but I still think this is relevant as you simply check if the entity exists.

Reply:
I'm not fond of this solution because

- it's unintuitive to me (I have to check from a different service whether is safe to call a method on the entity)
- it depends on the specific internal doctrine behavior (id is set on flush)

I would rather check for nullability of the getId method or introduce a new in the entity. 
We can discuss it in the broader circle

Reply:
I have removed both commits that were doing that